packages/combo-box/test/data-provider-filtering.common.js

```diff
@@ -0,0 +1,143 @@
+import { expect } from '@esm-bundle/chai';
+import { aTimeout, fixtureSync, nextRender } from '@vaadin/testing-helpers';
+import sinon from 'sinon';
+import { setInputValue } from './helpers.js';
+
+const TEMPLATES = {
+  'vaadin-combo-box': `
+    <vaadin-combo-box></vaadin-combo-box>
+  `,
+  'vaadin-combo-box-light': `
+    <vaadin-combo-box-light>
+      <vaadin-text-field></vaadin-text-field>
+    </vaadin-combo-box-light>
+  `,
+};
+
+['vaadin-combo-box', 'vaadin-combo-box-light'].forEach((tag) => {
+  describe(`data provider filtering - ${tag}`, () => {
+    let comboBox;
+
+    describe('basic', () => {
+      beforeEach(async () => {
+        comboBox = fixtureSync(TEMPLATES[tag]);
+        await nextRender();
+
+        comboBox.dataProvider = sinon.spy(async (params, callback) => {
+          const items = ['Item 1', 'Item 2', 'Item 3'];
+          const filteredItems = items.filter((item) => item.includes(params.filter));
+          await aTimeout(0);
+          callback(filteredItems, filteredItems.length);
+        });
+        comboBox.opened = true;
+        await aTimeout(0);
+      });
+
+      it('should make request with empty filter when opened', () => {
+        const params = comboBox.dataProvider.lastCall.args[0];
+        expect(params.filter).to.equal('');
+      });
+
+      it('should make request with entered filter on filter change', () => {
+        comboBox.dataProvider.resetHistory();
+        setInputValue(comboBox, 'Item');
+        expect(comboBox.dataProvider).to.be.called;
+        const params = comboBox.dataProvider.lastCall.args[0];
+        expect(params.filter).to.equal('Item');
+      });
+
+      it('should make request with empty filter on value change', async () => {
+        setInputValue(comboBox, 'Item');
+        await aTimeout(0);
+        comboBox.dataProvider.resetHistory();
+        comboBox.value = 'Item 2';
+        const params = comboBox.dataProvider.lastCall.args[0];
+        expect(params.filter).to.equal('');
+      });
+
+      it('should make request with empty filter after partial filter & cancel & reopen', async () => {
+        setInputValue(comboBox, 'It');
+        await aTimeout(0);
+        comboBox.dataProvider.resetHistory();
+        comboBox.cancel();
+        comboBox.opened = true;
+        await aTimeout(0);
+        const params = comboBox.dataProvider.lastCall.args[0];
+        expect(params.filter).to.equal('');
+      });
+
+      it('should clear filteredItems on filter change', () => {
+        setInputValue(comboBox, 'Item');
+        expect(comboBox.filteredItems).to.deep.equal([]);
+      });
+
+      it('should update filteredItems on filter request response', async () => {
+        setInputValue(comboBox, 'Item 1');
+        await aTimeout(0);
+        expect(comboBox.filteredItems).to.deep.equal(['Item 1']);
+      });
```

Comment:
**note:** I added these tests to cover https://github.com/vaadin/web-components/pull/7044#discussion_r1611331945